lib/phoenix_html/form.ex

```diff
@@ -391,7 +397,7 @@ defmodule Phoenix.HTML.Form do
         "password" => :password_input}
 
   """
-  @spec input_type(t, atom) :: atom
+  @spec input_type(t, atom | String.t) :: atom
```

Comment:
We should create a type named `field`, so we don't repeat `atom | String.t` all around the code.